### PR TITLE
Use Apache Commons Collections utility methods

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
The Apache Commons Collections library includes utility classes that perform standard collection operations in a null-safe way and also provide methods that return an empty `Collection` if the value passed in was null.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java [hunk 1]
```diff
@@ -73,7 +77,7 @@ private void showMenu() {
         menu.getMenuInflater().inflate(menuRes, menu.getMenu());
         menu.setOnMenuItemClickListener(new PageSaveMenuClickListener());
 
-        if (listsContainingPage.size() == 1) {
+        if (size(listsContainingPage) == 1) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java:80]: Why do we need a third-party library to get the size of a list?

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java [hunk 2]
```diff
@@ -82,16 +86,16 @@ private void showMenu() {
             menu.setGravity(Gravity.END);
 
             MenuItem addToOtherItem = menu.getMenu().findItem(R.id.menu_add_to_other_list);
-            addToOtherItem.setVisible(listsContainingPage.size() > 0);
-            addToOtherItem.setEnabled(listsContainingPage.size() > 0);
+            addToOtherItem.setVisible(isNotEmpty(listsContainingPage));
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java:89]: I could see perhaps changing this to `!list.isEmpty()`, but what is the benefit of using a new third-party dependency for this?

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java:80]: That particular list is nullable, so calling its size() method directly might throw a `NullPointerException`.

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java (hunk 2 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/readinglist/ReadingListBookmarkMenu.java:89]: Same as earlier: that list is nullable, so calling its `size()` method might throw a `NullPointerException`.

**Pull request closed without merging**